### DISCUSSION TITLE
Idea: Manage appveyor.yml by branch

### DISCUSSION THREAD

**Discussion by craigfowler** [elsa-workflows/elsa-core#660] [Design & tech]
Currently, the `appveyor.yml` file that's checked into source control (across all active long-lived branches) has the build configuration for _all of the branches_.  For example in the v2 branch the file has build configs for 1.4.x and 1.5.x as well as the 2.x code.  Which part of the yml build config is then chosen by syntax like:

```yml
- branches:
    only:
      - feature/elsa-2.0
```

This could be simplified though.  When AppVeyor runs a build, it uses the appveyor.yml file _from the same commit as it is building_.  This means that the file only actually needs to contain the config for the codebase that's in that same branch.  The (sort-of) YML switch/case that's going on isn't actually needed.

I think this would make the build config more clear and easier to maintain.  Have I missed perhaps a motivation for doing it the current way?  Any other thoughts/opinions?

**Comment by sfmskywalker** [comment]
This sounds good to me. The only reason it was done the way it is now is due to lack of experience with AppVeyor. Please feel free to go ahead and implement the suggested changes. Let me know if you need help with this.

**Reply by craigfowler** [reply]
Thanks, raised #664
